Populate listener_arn for aws_alb_listener_rule

sj26 opened the pull request:
Importing a listener rule currently doesn't populate the listener_arn so the next plan attempts to recreate the rule. The AWS API doesn't return the listener arn when describing listener rules. But the listener arn can be inferred from the listener rule arn.

Closes #779

radeksimko commented on aws/resource_aws_alb_listener_rule.go:
```diff
@@ -147,6 +148,15 @@ func resourceAwsAlbListenerRuleRead(d *schema.ResourceData, meta interface{}) er
+	// arn:aws:elasticloadbalancing:us-east-1:012345678912:listener/app/name/0123456789abcdef/abcdef0123456789/456789abcedf1234
+	listenerArn = listenerArn[0:strings.LastIndex(listenerArn, "/")]
+	// arn:aws:elasticloadbalancing:us-east-1:012345678912:listener/app/name/0123456789abcdef/abcdef0123456789
+	d.Set("listener_arn", listenerArn)
```
This seems functionally ok, but do you mind decoupling this new logic into a function and adding a test for it?

sj26 commented:
@radeksimko thanks for the feedback! Sorry, I was away for a few days. I've extracted that logic into a little function and added some testing modelled after the [alb target group cloudwatch suffix tests](https://github.com/terraform-providers/terraform-provider-aws/blob/4f9196eaf93734078a986decf859f8f995c25383/aws/resource_aws_alb_target_group_test.go#L17-L46). I also switched to using a regexp because it wasn't handling the empty case gracefully.
